src/source.rs

```diff
@@ -66,21 +71,70 @@ pub enum LoadResponse {
 pub type LoadResult = Result<Option<LoadResponse>>;
 pub type LoadFuture = LocalBoxFuture<'static, LoadResult>;
 
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub enum LoaderCacheSetting {
+  /// Attempts to load a specifier from the cache.
+  ///
+  /// This is used to see whether the specifier is in the cache for `deno:` specifiers.
+  /// * If it is, then it will use the source provided to get the module information.
+  /// * If not, then it will use the manifest information to do resolution and
+  ///   issue a separate request to the `load` method in order to get the source.
+  Only,
+  /// The implementation should prefer using the cache.
+  Prefer,
+  /// Loads a specifier where the implementation should not load
+  /// from an internal cache. This is only ever done when loading
+  /// `deno:` specifier module information and the version constraint
+  /// cannot be resolved.
+  Reload,
+}
+
+pub static DEFAULT_DENO_REGISTRY_URL: Lazy<Url> =
+  Lazy::new(|| Url::parse("https://registry-staging.deno.com").unwrap());
```

Comment:
Maybe this should be injected when constructing the graph?

Reply:
It can be already by providing it on the loader